controllers/serviceclaim_controller.go

```diff
@@ -343,6 +411,55 @@ func (r *ServiceClaimReconciler) pushToClusterEnvironments(
 	return nil
 }
 
+func (r *ServiceClaimReconciler) DeleteServiceBindingsAndSecret(
+	ctx context.Context,
+	req ctrl.Request,
+	sclaim primazaiov1alpha1.ServiceClaim,
+) error {
+	l := log.FromContext(ctx)
+	errs := []error{}
+
+	if sclaim.Spec.ApplicationClusterContext != nil {
+		var err error
+		ce, err := r.getEnvironmentFromClusterEnvironment(ctx, req, sclaim.Spec.ApplicationClusterContext.ClusterEnvironmentName)
+		if err != nil {
+			l.Info("error getting ClusterEnvironment", "error", err)
+			return err
+		}
+		cli, err := clustercontext.CreateClient(ctx, r.Client, *ce, r.Scheme, r.Client.RESTMapper())
+		if err != nil {
+			return err
+		}
+		if err = controlplane.DeleteServiceBindingAndSecretFromNamespaces(ctx, cli, sclaim, ce.Spec.ApplicationNamespaces); err != nil {
```

Comment:
The ApplicationClusterContext defines also the target namespace

```suggestion
                ns := []string{sclaim.Spec.ApplicationClusterContext.Namespace}
		if err = controlplane.DeleteServiceBindingAndSecretFromNamespaces(ctx, cli, sclaim, ns); err != nil {
```